### PR TITLE
update KWS for the Blackboard.postArtifacts API

### PR DESCRIPTION
**Pull request by millmanorama**
comes after PR 4040

### CODE REVIEW

**Comment by millmanorama** [comment]
i address all the non-TODO related comments.  I had hoped to get some feedback from @bcarrier or @rcordovano about the TODOs and not have to write up stories.  If that is not going to happen I will make jira stories for them.

File: RecentActivity/src/org/sleuthkit/autopsy/recentactivity/Chrome.java [hunk 1]
```diff
@@ -107,11 +127,12 @@ private void getHistory() {
 
         // log a message if we don't have any allocated history files
         if (allocatedHistoryFiles.isEmpty()) {
-            String msg = NbBundle.getMessage(this.getClass(), "Chrome.getHistory.errMsg.couldntFindAnyFiles");
+            String msg = NbBundle.getMessage(Chrome.class, "Chrome.getHistory.errMsg.couldntFindAnyFiles");
             logger.log(Level.INFO, msg);
             return;
         }
 
+        //TODO  why are we using historyFiles instead of allocatedHistoryfiles here?
```
[the remark this answers is not in the document]
Comment (bcarrier) [reply RecentActivity/src/org/sleuthkit/autopsy/recentactivity/Chrome.java]: This seems to be an incorrect fix to a problem associated with duplicate artifacts.  I created JIRA-4158 to properly deal with duplicate web artifacts.

File: RecentActivity/src/org/sleuthkit/autopsy/recentactivity/RecentDocumentsLnkExtractor.java [hunk 2]
```diff
@@ -96,39 +114,42 @@ private void getRecentDocuments() {
...
-                BlackboardArtifact.ARTIFACT_TYPE.TSK_RECENT_OBJECT));
-    }
 
-    @Override
-    public void process(Content dataSource, IngestJobContext context) {
-        this.dataSource = dataSource;
-        this.context = context;
-        dataFound = false;
-        this.getRecentDocuments();
+        //TODO: why weren't these getting indexed before?
```
Comment (bcarrier) [inline RecentActivity/src/org/sleuthkit/autopsy/recentactivity/RecentDocumentsLnkExtractor.java]: probably an oversight.  You can remove the TODO.

**Comment by bcarrier** [comment]
@millmanorama this now conflicts.

**Comment by millmanorama** [comment]
I fixed the conflicts

**Pull request merged**